_posts/2021-08-23-gsoc-2021-project.md

```diff
@@ -0,0 +1,208 @@
+
+Note: this change only affects the graphviz output for Dask high level graphs. Low level graphs are left unchanged.
+
+Reference: [Pull request #7869 by Freyam Mehta *"Add node size scaling to the Graphviz output for the high level graphs"*](https://github.com/dask/dask/pull/7869)
+
+### New tooltips
+
+Dask high level graphs now include hover tooltips to provide a brief summary of more detailed information. To use the tooltips, generate a dask high level graph (eg: `result.dask.visualize()`) then hover your mouse above the layer you are interested in.
+
+<img src="/images/gsoc21/7973.png" alt="Example: tooltips provide extra information, pull request #7973" height=414 width=736>
+
+Tooltips provide information such as the layer type and number of tasks associated with it. There is additional information provided for specific dask collections, like dask arrays and dataframes.
+
+Dask array tooltip information additionally includes:
+* Array shape
+* Chunk size
+* Chunk type (eg: are the array chunks numpy, cupy, sparse, etc.)
+* Data type (eg: are the array values float, integer, boolean, etc.)
+
+Dask dataframe tooltip information additionally includes:
+* Number of partitions
+* Dataframe type
+* Dataframe columns
+
+Users have asked for a less overwhelming view into the dask task graph. We hope the high level graph view coupled with more detailed tooltip information can provide this middle ground, with enough information to be useful, but not so much as to become overwhelming (like the low level task graphs for large computations).
+
+Note: This feature is available for SVG output. Other image formats, like `.png`, etc. do not support tooltips.
+
+Reference: [Pull request #7973 by Freyam Mehta *"Add tooltips to graphviz"*](https://github.com/dask/dask/pull/7973)
+
+### Color by layer type
+
+There is also a new feature enabling users to color code a high level graph according to layer type. This option can be enabled by passing the `color="layer_type"` keyword argument, eg: `result.dask.visualize(color="layer_type")`. This change is intended to make it easier for users to see which layer types predominate.
+
+While there are no hard and fast rules about what makes a Dask computation efficient, there are some general guidelines:
+
+* Dataframe shuffles are particularly expensive operations. You can [read more about this here](https://docs.dask.org/en/latest/dataframe-best-practices.html#avoid-full-data-shuffling).
+* Reading and writing data to/from disk can be a limiting factor. (Similarly, you want to avoid Dask spilling data to disk wherever possible - monitor the worker memory plot on the dask dashboard if you suspect this may be happening).
+* Blockwise layers are generally efficient for computation.
+
+See the [Dask best pracices](https://docs.dask.org/en/latest/best-practices.html) pages for more information on creating more efficient Dask computations.
+
+Example:
+
+```python
+# code
```

Comment:
@freyam please fill in the example code you used to generate the demo images below.